### ISSUE TITLE
Assertion fail when insert and delete in the same transaction

### ISSUE THREAD

**Issue by nwang57** [cmu-db/peloton#1336]
```
create table foo(id integer);

begin;
insert into foo values(5);
delete from foo where id = 5;
```
The assertion failed at line 618 in `src/concurrency/timestamp_ordering_transaction_manager.cpp`.  It seems that if the txn owns the tuple, it will not update the tuple's last reader id. So the commitId is 0 during the assertion.

**Comment by apavlo** [comment]
Thank you for reporting this. We are having a lot of problems with our concurrency control scheme. Is this a blocker for your project?

**Comment by nwang57** [comment]
Yeah, we passed all the `make check` tests except the functions_test which is related to this problem. We will talk with @pmenon on Friday about this issue.

BYW, we just merged with catalog refactoring commit in the master with our changes. So far it doesn't break any test yet.
